### PR TITLE
Use Token SDK in a sample cordapp of cross-chain atomic swap

### LINKED ISSUE #13: Use Token SDK in a sample of cross-chain atomic swap application
Update `SecurityState` into the evolvable fungible token.

Token SDK requires Corda 4.3 or later.

### LINKED ISSUE #26: Several contract tests don't pass
スクリーンショット 2020-04-18 17 16 41

These failures are weird.

When running `input proposal's proposer must equal to output security state's owner` test,
```
ledgerServices.ledger {
    transaction {
        input(ProposalContract.contractID, inputProposal)
        input(SecurityContract.contractID, inputSecurity)
        output(ProposalContract.contractID, outputProposal)
        output(SecurityContract.contractID, outputSecurity)
        command(listOf(ALICE.publicKey, BOB.publicKey), ProposalContract.ProposalCommands.Consume())
        command(listOf(BOB.publicKey, ALICE.publicKey, CHARLIE.publicKey), SecurityContract.SecurityCommands.TransferForSettle())
        this `fails with` "InputProposalState's proposer must equal to OutputSecurityState's owner."
    }
}
```

the transaction seems to have 2 inputs, 2 outputs, and 2 commands but it has only 1 output.

スクリーンショット 2020-04-18 17 41 44

### PR DESCRIPTION
**Pull request by shun-tak**
This resolves #13
This fixes #26

### CODE REVIEW

File: cross-chain-atomic-swap-cordapp/src/main/kotlin/jp/co/layerx/cordage/crosschainatomicswap/contract/ProposalContract.kt [hunk 1]
```diff
@@ -36,12 +35,8 @@ open class ProposalContract: Contract {
                     (proposalCommand.signers.toSet() == proposal.participants.map { it.owningKey }.toSet())
             }
             is ProposalCommands.Consume -> requireThat {
```
Comment (etaroid) [inline cross-chain-atomic-swap-cordapp/src/main/kotlin/jp/co/layerx/cordage/crosschainatomicswap/contract/ProposalContract.kt:37]: Isn't it necessary to verify whether the content of the ProposalState and the content of the Bond being transferred are the same here?

This is a validation that was previously written in SecurityContract.

File: cross-chain-atomic-swap-cordapp/README.md [hunk 2]
```diff
@@ -41,48 +41,59 @@ See https://docs.corda.net/tutorial-cordapp.html#running-the-example-cordapp.
...
-### vaultQuery for Security State
-Run vaultQuery from ParticipantB:
+Every Data propagation between Corda and Ethereum is executed by Corda Nodes or Notary.
+Other processes are not required.
 
+### Setup
+#### Register CorporateBond from Party C
 ```
-run vaultQuery contractStateType: jp.co.layerx.cordage.crosschainatomicswap.state.SecurityState
+flow start jp.co.layerx.cordage.crosschainatomicswap.flow.CorporateBondRegisterFlow name: "LayerX", unitPriceEther: "0.012345678901234567", observer: "O=ParticipantA,L=London,C=GB"
```
[the remark this answers is not in the document]
Comment (shun-tak) [reply cross-chain-atomic-swap-cordapp/README.md:72]: The observer parameter means just sharing a CorporateBond state to a designated node once. not every time.

When Party A starts ProposalAtomcSwapFlow, she needs to know the CorporateBond state so Party A is shared by using the observer parameter.

If there exist more nodes, you can share the state in the ad-hoc way by using `session.receive`
https://docs.corda.net/docs/corda-os/4.4/api-flows.html#receive

File: cross-chain-atomic-swap-cordapp/src/main/kotlin/jp/co/layerx/cordage/crosschainatomicswap/contract/ProposalContract.kt (hunk 1 above)
Comment (shun-tak) [reply cross-chain-atomic-swap-cordapp/src/main/kotlin/jp/co/layerx/cordage/crosschainatomicswap/contract/ProposalContract.kt:37]: it slipped in my mind. I'll add the constraint.

File: cross-chain-atomic-swap-cordapp/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (etaroid) [reply cross-chain-atomic-swap-cordapp/README.md:72]: I got it. 
LGTM!

File: cross-chain-atomic-swap-cordapp/src/main/kotlin/jp/co/layerx/cordage/crosschainatomicswap/flow/SettleAtomicSwapFlow.kt [hunk 3]
```diff
@@ -109,9 +118,14 @@ class SettleAtomicSwapFlowResponder(val flowSession: FlowSession) : FlowLogic<Si
     override fun call(): SignedTransaction {
         val signedTransactionFlow = object : SignTransactionFlow(flowSession) {
             override fun checkTransaction(stx: SignedTransaction) = requireThat {
-                // add any validation by yourself
-                // val securityOutput = stx.tx.outputsOfType<SecurityState>().first()
-                // val proposalOutput = stx.tx.outputsOfType<ProposalState>().first()
+                val proposalState = stx.tx.outputsOfType<ProposalState>().single()
+                val fungibleTokens = stx.tx.outputsOfType<FungibleToken>()
+                requireThat {
+                    "ourIdentity's address must equal to fromEthereumAddress." using (proposalState.fromEthereumAddress == ourIdentity.ethAddress())
```
[the remark this answers is not in the document]
Comment (shun-tak) [reply cross-chain-atomic-swap-cordapp/src/main/kotlin/jp/co/layerx/cordage/crosschainatomicswap/flow/SettleAtomicSwapFlow.kt:124]: it is validated soon after this line

https://github.com/LayerXcom/cordage/blob/a07dd2473e3cf7ff331e0c6ed79a2405cdde5d9f/cross-chain-atomic-swap-cordapp/src/main/kotlin/jp/co/layerx/cordage/crosschainatomicswap/flow/SettleAtomicSwapFlow.kt#L75-L77

File: cross-chain-atomic-swap-cordapp/src/main/kotlin/jp/co/layerx/cordage/crosschainatomicswap/flow/ProposeAtomicSwapFlow.kt [hunk 4]
```diff
@@ -119,19 +124,16 @@ class ProposeAtomicSwapFlowResponder(val flowSession: FlowSession) : FlowLogic<S
...
         val signedTransactionFlow = object : SignTransactionFlow(flowSession) {
             override fun checkTransaction(stx: SignedTransaction) = requireThat {
-                val output = stx.tx.outputs.single().data
-                "This must be an Proposal transaction" using (output is ProposalState)
-                // add any validation by yourself
+                val proposalState = stx.tx.outputs.single().data
+                "This must be an Proposal transaction" using (proposalState is ProposalState)
+                proposalState as ProposalState
+                "Proposer's address must equal to fromEthereumAddress." using (proposalState.fromEthereumAddress == proposalState.proposer.ethAddress())
+                "ourIdentity's address must equal to toEthereumAddress." using (proposalState.toEthereumAddress == ourIdentity.ethAddress())
```
[the remark this answers is not in the document]
Comment (shun-tak) [reply cross-chain-atomic-swap-cordapp/src/main/kotlin/jp/co/layerx/cordage/crosschainatomicswap/flow/ProposeAtomicSwapFlow.kt:131]: that's true! we should validate that

File: cross-chain-atomic-swap-cordapp/src/main/kotlin/jp/co/layerx/cordage/crosschainatomicswap/flow/SettleAtomicSwapFlow.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (shun-tak) [reply cross-chain-atomic-swap-cordapp/src/main/kotlin/jp/co/layerx/cordage/crosschainatomicswap/flow/SettleAtomicSwapFlow.kt:124]: that's true! we should validate that

**Pull request merged**